### PR TITLE
Update docs - obtaining references 

### PR DESCRIPTION
**Pull request by helena**
Call out locality, slightly clarify and note context and origin from Carl Hewitt's paper. Thoughts while digging into receptionist / pubsub.

### CODE REVIEW

File: akka-docs/src/main/paradox/typed/actor-discovery.md [hunk 1]
```diff
@@ -14,29 +14,37 @@ To use Akka Actor Typed, you must add the following dependency in your project:
...
-different nodes in the Cluster or when "dependency injection" with constructor parameters is not
-applicable.
+In order for an actor to send a message to one or more other actors it needs to obtain an @ref:[Actor reference](../general/addressing.md#what-is-an-actor-reference). 
+This reference relates to the notion of locality.
+In his 2010 paper, [Actor Model of Computation: Scalable Robust Information Systems](https://arxiv.org/abs/1008.1459), 
+Carl Hewitt relays locality and message processing as, 
+“an Actor can send messages only to addresses for which it has information by the following means: 
+1. that it receives in the message 
+2. that it already had before it received the message 
+3. that it creates while processing the message. “
```
Comment (johanandren) [inline akka-docs/src/main/paradox/typed/actor-discovery.md]: I'm not convinced we should go into theory here, I like the old hands on style better. Is there no there place in the docs where we do some more theory/background section where we already mention this that we could potentially crosslink instead?

File: akka-docs/src/main/paradox/typed/actor-discovery.md [hunk 2]
```diff
@@ -14,29 +14,37 @@ To use Akka Actor Typed, you must add the following dependency in your project:
...
 
 ## Receptionist
 
-When an actor needs to be discovered by another actor but you are unable to put a reference to it in an incoming message,
-you can use the `Receptionist`. You register the specific actors that should be discoverable 
-from other nodes in the local `Receptionist` instance. The API of the receptionist is also based on actor messages. 
-This registry of actor references is then automatically distributed to all other nodes in the cluster. 
-You can lookup such actors with the key that was used when they were registered. The reply to such a `Find` request is 
-a `Listing`, which contains a `Set` of actor references that are registered for the key. Note that several actors can be 
+The `Receptionist` is useful when specific actors need to be discoverable by other actors and you need to bootstrap the interaction.
```
Comment (johanandren) [inline akka-docs/src/main/paradox/typed/actor-discovery.md]: WDYT about completely skipping "and you need to bootstrap the interaction"?

File: akka-docs/src/main/paradox/typed/actor-discovery.md [hunk 3]
```diff
@@ -14,29 +14,37 @@ To use Akka Actor Typed, you must add the following dependency in your project:
...
 ## Receptionist
 
-When an actor needs to be discovered by another actor but you are unable to put a reference to it in an incoming message,
-you can use the `Receptionist`. You register the specific actors that should be discoverable 
-from other nodes in the local `Receptionist` instance. The API of the receptionist is also based on actor messages. 
-This registry of actor references is then automatically distributed to all other nodes in the cluster. 
-You can lookup such actors with the key that was used when they were registered. The reply to such a `Find` request is 
-a `Listing`, which contains a `Set` of actor references that are registered for the key. Note that several actors can be 
+The `Receptionist` is useful when specific actors need to be discoverable by other actors and you need to bootstrap the interaction. 
+For example when they need to be discoverable by actors running either on different nodes in the Cluster,
```
Comment (johanandren) [inline akka-docs/src/main/paradox/typed/actor-discovery.md]: "when they need to be discoverable by" repeated in to sentences in a row, so maybe skip that here?

File: akka-docs/src/main/paradox/typed/actor-discovery.md [hunk 4]
```diff
@@ -14,29 +14,37 @@ To use Akka Actor Typed, you must add the following dependency in your project:
...
+The reply to a `Find` request is a `Receptionist.Listing` containing a `Set` of actor references registered for the key. Note that several actors can be 
 registered to the same key.
 
 The registry is dynamic. New actors can be registered during the lifecycle of the system. Entries are removed when 
-registered actors are stopped, manually deregistered or the node they live on is removed from the @ref:[Cluster](cluster.md). 
-To facilitate this dynamic aspect you can also subscribe to changes with the `Receptionist.Subscribe` message. It will send 
-`Listing` messages to the subscriber when entries for a key are changed.
+registered actors are stopped, manually de-registered or the node they live on is removed from the @ref:[Cluster](cluster.md). 
+Because it is dynamic, changes for a`ServiceKey` can be subscribed to with the `Receptionist.Subscribe` message.
+and a `Receptionist.Listing` message is sent to those subscribers as changes occur.
```
Comment (johanandren) [inline akka-docs/src/main/paradox/typed/actor-discovery.md]: We should probably mention `GroupRouter` immediately here, "Keeping track of available services for a key and routing messages to one of them, which is one of the most common use cases is covered by the @ref[GroupRouter] but there are use cases when keeping track of registered services are needed for more complex logic" something something

**Comment by eloots** [comment]
@helena
The last sentence of the last paragraph of `actor-discovery.md` looks garbled.
Did you mean this?
```
Use cases with higher demands can utilise the receptionist for establishing initial contact between actors on the nodes while the actual logic of those is up to the applications own actors.
```

**Pull request closed without merging**